File: java/src/main/java/api/cornell/SubjectsResponse.java
package api.cornell;

import api.cornell.data.classes.Subject;
import org.jetbrains.annotations.Nullable;

import java.util.List;

/**
 * {@code SubjectsResponse} is the response of a Subjects request.
 */
final class SubjectsResponse {
    
    /**
     * Main data of response.
     */
    @Nullable
    private final Data data;
    
    /**
     * Constructor for GSON.
     */
    private SubjectsResponse() {
        data = null;
    }
    
    /**
     * Obtain the subjects information.
     *
     * @return the subjects information.
     */
    @Nullable
    public List<Subject> getSubjects() {
        if (data == null) {
            return null;
        }
        return data.subjects;
    }
    
    /**
     * {@code Data} is an uninteresting class that just holds an array of {@code Page}.
     */
    private static class Data {
        
        @Nullable
        private List<Subject> subjects;
    
        /**
         * Constructor for GSON.
         */
        private Data() {
            subjects = null;
        }
        
    }
}

File: kotlin/src/main/kotlin/api/cornell/response/classes/SubjectsResponse.kt
package api.cornell.response.classes

import api.cornell.data.classes.Subject

/**
 * [SubjectsResponse] is the response of a Subjects request.
 */
internal class SubjectsResponse private constructor() {

    /**
     * Main data of response.
     */
    private val data: Data = Data()
    /**
     * Obtain the class levels information.
     */
    val subjects: List<Subject> get() = data.subjects

    /**
     * [Data] is an uninteresting class that just holds an array of [Subject].
     */
    private inner class Data internal constructor() {
        val subjects: List<Subject> = emptyList()
    }

}


